org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/TerminalTokens.java

```
@@ -71,143 +235,18 @@ static int getRestrictedKeyword(String text) {
-							TokenNameREMAINDER_EQUAL = 100,
-							TokenNameLEFT_SHIFT_EQUAL = 101,
-							TokenNameRIGHT_SHIFT_EQUAL = 102,
-							TokenNameUNSIGNED_RIGHT_SHIFT_EQUAL = 103,
-							TokenNameOR_OR = 31,
-							TokenNameAND_AND = 30,
-							TokenNamePLUS = 4,
-							TokenNameMINUS = 5,
-							TokenNameNOT = 67,
-							TokenNameREMAINDER = 9,
-							TokenNameXOR = 25,
-							TokenNameAND = 21,
-							TokenNameMULTIPLY = 8,
-							TokenNameOR = 28,
-							TokenNameTWIDDLE = 68,
-							TokenNameDIVIDE = 10,
-							TokenNameGREATER = 15,
-							TokenNameLESS = 11,
-							TokenNameLPAREN = 23,
-							TokenNameRPAREN = 24,
-							TokenNameLBRACE = 63,
-							TokenNameRBRACE = 33,
-							TokenNameLBRACKET = 6,
-							TokenNameRBRACKET = 70,
-							TokenNameSEMICOLON = 26,
-							TokenNameQUESTION = 29,
-							TokenNameCOLON = 66,
-							TokenNameCOMMA = 32,
-							TokenNameDOT = 1,
-							TokenNameEQUAL = 78,
-							TokenNameAT = 37,
-							TokenNameELLIPSIS = 128,
-							TokenNameARROW = 118,
-							TokenNameCOLON_COLON = 7,
-							TokenNameBeginLambda = 64,
-							TokenNameBeginIntersectionCast = 69,
-							TokenNameBeginTypeArguments = 89,
-							TokenNameElidedSemicolonAndRightBrace = 73,
-							TokenNameAT308 = 27,
-							TokenNameAT308DOTDOTDOT = 135,
-							TokenNameCaseArrow = 74,
-							TokenNameRestrictedIdentifierYield = 90,
-							TokenNameRestrictedIdentifierrecord = 76,
-							TokenNameRestrictedIdentifiersealed = 42,
-							TokenNameRestrictedIdentifierpermits = 129,
-							TokenNameBeginCasePattern = 130,
-							TokenNameRestrictedIdentifierWhen = 131,
-							TokenNameUNDERSCORE = 34,
-							TokenNameEOF = 39,
-							TokenNameERROR = 138;
+	public static TerminalTokens of(int act) {
+		TerminalTokens token = tokenMap.get(act);
+		if (token == null) {
+			// Really shouldn't occur -- perhaps if parser non-terminals are looked up?
+			throw new IllegalArgumentException("Unknown token number = " + act); //$NON-NLS-1$
```

Rather than throw, return `TokenNameNotAToken`?

---

I'm afraid that it would mask an error during the refactor. Or after.

---

> I'm afraid that it would mask an error during the refactor. Or after.

I am not super particular with the suggested change. Feel free to leave it as is